### DISCUSSION TITLE
Specify Generated Input Arguments As Required

### DISCUSSION THREAD

**Discussion by gblikas** [MichalLytek/typegraphql-prisma#337] [Q&A]
In the example code on https://prisma.typegraphql.com/docs/basics/usage,

```
@TypeGraphQL.ObjectType({
  isAbstract: true,
  description: undefined,
})
export class User {
  @TypeGraphQL.Field(_type => String, {
    nullable: false,
    description: undefined,
  })
  id!: string;

  @TypeGraphQL.Field(_type => String, {
    nullable: false,
    description: undefined,
  })
  email!: string;

  @TypeGraphQL.Field(_type => Int, {
    nullable: true,
    description: undefined,
  })
  age?: number | null;

  @TypeGraphQL.Field(_type => UserKind, {
    nullable: false,
    description: undefined,
  })
  kind!: UserKind;
}
```

`id` is a required field. However, when looking at an auto-generated Input type like `UserWhereUniqueInput`, the `id` field isn't required, 

```
// ... 
id?: string | undefined 
// ... 
```

is there a way to force typegraphql-prisma to make the field required on (generated) Input types, as well?

**Comment by MichalLytek** [comment]
No, generated inputs match Prisma Client API.

**Reply by gblikas** [reply]
> No, generated inputs match Prisma Client API.

@MichalLytek I don't get it. In the above case, wouldn't what your saying mean 
```
id: string
```

**Reply by MichalLytek** [reply]
I mean all the input types definition are provided by Prisma DMMF, which maps 1:1 the Prisma Client API - the TS types for `prisma.user.findUnique()`. So when you call Prisma findUnique, `id` field is not required, so it's nullable in `UserWhereUniqueInput`.

**Reply by gblikas** [reply]
Any opinions on making this part adaptable to similar use-cases, like mine?
